pkg/designate/pool_update.go

```diff
@@ -0,0 +1,132 @@
+package designate
+
+import (
+	"fmt"
+
+	designatev1beta1 "github.com/openstack-k8s-operators/designate-operator/api/v1beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	PoolUpdateCommand      = "/usr/bin/designate-manage pool update"
+	DesignateConfigVolume  = "designate-config"
+	DesignateConfigMount   = "/etc/designate"
+	DesignateConfigKeyPath = "designate.conf"
+	DesignatePoolsYamlPath = "pools.yaml"
+)
+
+func PoolUpdateJob(
+	instance *designatev1beta1.Designate,
+	labels map[string]string,
+	annotations map[string]string,
+) *batchv1.Job {
+	runAsUser := int64(0)
+	volumeMounts := GetVolumeMounts(instance.Name)
+	volumes := GetVolumes(instance.Name)
+
+	volumes = append(volumes, corev1.Volume{
+		Name: "pools-yaml-config",
+		VolumeSource: corev1.VolumeSource{
+			ConfigMap: &corev1.ConfigMapVolumeSource{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: PoolsYamlConfigMap,
+				},
+				Items: []corev1.KeyToPath{
+					{
+						Key:  "pools-yaml-content",
+						Path: DesignatePoolsYamlPath,
+					},
+				},
+			},
+		},
+	})
+
+	volumes = append(volumes, corev1.Volume{
+		Name: DesignateConfigVolume,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: "designate-api-config-data",
```

Comment:
I wonder if this is the right secret name. The pool update job probably has more in common with central. I suggest we retest with the merged redis patch to make sure that this still works.

Reply:
Sounds good to me. I think that I implemented it before I had access to the redis PR. So now that we had an "upgraded" version of central, let me change it and test.